app/js/features/balances/saga.js

```diff
@@ -1,21 +1,40 @@
-import SNT from 'Embark/contracts/SNT';
-import {fork, takeEvery, put} from 'redux-saga/effects';
+/*global web3*/
+
+import ERC20Token from 'Embark/contracts/ERC20Token';
+import {fork, takeEvery, put, all} from 'redux-saga/effects';
 import {
-  LOAD_SNT_BALANCE_SUCCEEDED, LOAD_SNT_BALANCE_FAILED, LOAD_SNT_BALANCE
+  UPDATE_BALANCES, UPDATE_BALANCE, UPDATE_BALANCE_FAILED, UPDATE_BALANCE_SUCCEEDED
 } from './constants';
+import { zeroAddress } from '../../utils/address';
 
-export function *loadSNTBalance({address}) {
+export function *updateBalance({token, address}) {
   try {
-    const value = yield SNT.methods.balanceOf(address).call();
-    yield put({type: LOAD_SNT_BALANCE_SUCCEEDED, value: value, address: address});
+    let value;
+    if (token.address === zeroAddress) {
+      value = yield web3.eth.getBalance(address);
+    } else {
+      const contract = new web3.eth.Contract(ERC20Token._jsonInterface, token.address);
```

Comment:
There's a legit getter in `contract.options` that's called `jsonInterface` I think.